pkg/util/containers/filter.go

```diff
@@ -72,37 +79,62 @@ type Filter struct {
 	NamespaceExcludeList []*regexp.Regexp
 }
 
+// filterErrors holds a list of regex errors and warnings for invalid filters
+type filterErrors struct {
```

Comment:
I tried testing with the `Filter` struct, but if there is a regex parsing error, it look like `NewFilter()` will return `nil` and an error [here](https://github.com/DataDog/datadog-agent/pull/8481/files#diff-655a2fb6bf73aef9a3d353233d78436839850406e9f550cca25cfaece8f3804eR195-R199) so the regex errors aren't ever stored. Would it be better to change that section to return `&Filter{FilterErrors: mapoferrors}` instead of `nil` and use the `Filter` struct?

Reply:
yes we can do that! let's just name the field `Errors` to avoid the redundancy `Filter.FilterErrors` -> `Filter.Errors`